src/aiida_quantumespresso/tools/code_setup.py

```diff
@@ -0,0 +1,209 @@
+# -*- coding: utf-8 -*-
+"""Module for setting up `Code` instances for Quantum ESPRESSO executables."""
+
+import pathlib
+import re
+from typing import Union
+
+from aiida import orm
+from aiida.cmdline.utils import echo
+from aiida.common.exceptions import NotExistent
+import click
+
+
+def create_codes(
+    computer: orm.Computer,
+    executables: Union[str, list[str], tuple[str]],
+    directory: Union[str, None] = None,
+    label_template: str = '',
+    prepend_text: Union[str, None] = None,
+    append_text: Union[str, None] = None,
+    on_conflict: str = 'increment',
+):
+    """Automatically create `orm.Code` instances for Quantum ESPRESSO executables.
+
+    :param computer: the `orm.Computer` on which the executables are located
+    :param executables: Quantum ESPRESSO executable(s) to create code(s) for, e.g. `pw.x` or `['pw.x', 'projwfc.x']`
+    :param directory: absolute path to a directory where the executable(s) is (are) located
+                      If not provided, the command will try to find the executables in the `PATH`
+                      on the (remote) computer
+    :param label_template: label template for the code instance. Use curly brackets to reference
+                           the executable label, e.g. `qe-{}` will create a `Code` with label
+                           `qe-pw` for `pw.x`. Defaults to the executable name without `.x` suffix.,
+                           e.g. `pw` for `pw.x`.
+    :param prepend_text: text to prepend to the execution command, e.g. to load modules
+    :param append_text: text to append to the execution command
+    :param on_conflict: action to take if a code with the same label already exists on the computer. Can be one of:
+                        - 'increment': append an incremental suffix to the label, e.g. `pw-2` if `pw` already exists
+                        - 'skip': skip creating the code for that executable
+                        - 'prompt': prompt the user to confirm whether to create another code with an incremented
+                          label or skip creating the code for that executable
+    """
+    if isinstance(executables, str):
```

Comment:
I don't fully understand the need for this wrapper, making the code more dispersed. Can't you simply do this validation in `_create_codes`?